### PR TITLE
Include metadata.json with each DB

### PR DESCRIPTION
**Pull request by simar7**
This includes a `metadata.json` file for when the DB was built with the asset.

I've decided to include metadata.json with each type of DB so the user can download any (full or light) for an air-gapped / manual download and it would still work. I think this is cleaner rather than maintaining yet another DB type since the `metadata.json` barely takes any space (107 B)

Addresses: https://github.com/goharbor/harbor/issues/11985

Signed-off-by: Simarpreet Singh <simar@linux.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,13 +78,20 @@ db-build: trivy-db
...
+	mkdir -p assets/$(DB_TYPE)
+	$(GOBIN)/bbolt compact -o ./assets/$(DB_TYPE)/$(DB_TYPE).db cache/db/trivy.db
+	cp cache/db/metadata.json ./assets/$(DB_TYPE)/metadata.json
 	rm cache/db/trivy.db
 
 .PHONY: db-compress
-db-compress: assets/$(DB_TYPE).db
-	gzip assets/*
+db-compress: assets/trivy/trivy.db
+	zip -9 assets/trivy/trivy.db.gz assets/trivy/trivy.db assets/trivy/metadata.json
```
[the remark this answers is not in the document]
Comment (simar7) [reply Makefile]: Actually this is rather unfortunate. `gzip` is just a compressor not an archiver so there's no way to package multiple like we want to do here. 

I have another idea, where we can filter out a separate DB (one for each full and light). This is similar to what you had alluded to doing earlier. https://github.com/aquasecurity/trivy-db/pull/47/commits/dd2cd6f69d6b55e64ada43e8dc820db234cd4f85

So ultimately, we will upload 4 artifacts:
```
-rw-------  1 simar  staff    16M Jun  1 14:55 trivy.db.gz
-rw-r--r--  1 simar  staff    17M Jun  1 15:17 trivy-offline.db.tgz
-rw-------  1 simar  staff   4.3M Jun  1 15:25 trivy-light.db.gz
-rw-r--r--  1 simar  staff   4.6M Jun  1 15:25 trivy-light-offline.db.tgz
```

File: Makefile [hunk 2]
```diff
@@ -78,13 +78,22 @@ db-build: trivy-db
...
+	cp cache/db/metadata.json ./assets/$(DB_TYPE)/metadata.json
 	rm cache/db/trivy.db
 
 .PHONY: db-compress
-db-compress: assets/$(DB_TYPE).db
-	gzip assets/*
+db-compress: assets/trivy/trivy.db
+	tar -cvzf assets/trivy-offline.db.tgz assets/trivy/trivy.db assets/trivy/metadata.json
+	gzip --best assets/trivy/trivy.db
+	mv assets/trivy/trivy.db.gz assets/trivy.db.gz
```
[the remark this answers is not in the document]
Comment (simar7) [reply Makefile]: I fixed here https://github.com/aquasecurity/trivy-db/pull/47/commits/cc445bddf03d618c21396c5f583ff54a4547d1d3#diff-b67911656ef5d18c4ae36cb6741b7965R87-R90

File: pkg/db/db.go [hunk 3]
```diff
@@ -130,6 +131,14 @@ func (dbc Config) SetMetadata(metadata Metadata) error {
 	return nil
 }
 
+func (dbc Config) StoreMetadata(metadata Metadata, dir string) error {
+	b, err := json.Marshal(metadata)
+	if err != nil {
+		return err
```
[the remark this answers is not in the document]
Comment (simar7) [reply pkg/db/db.go]: https://github.com/aquasecurity/trivy-db/pull/47/commits/518d0406bcc6e39145e2e6c2187f32a031029d1c

File: .github/workflows/cron.yml [hunk 4]
```diff
@@ -43,6 +43,11 @@ jobs:
     - name: Compact DB
       run: make db-compact
 
+    - name: Compress assets
+      run: make db-compress
+      env:
+        DB_TYPE: trivy-light
```
Comment (simar7) [inline .github/workflows/cron.yml:49]: I think you didn't meant this to be here right?

Comment (knqyf263) [reply .github/workflows/cron.yml]: oooops. sorry for my mistake

Comment (simar7) [reply .github/workflows/cron.yml]: No worries! I fixed it: https://github.com/aquasecurity/trivy-db/pull/47/commits/09f3033bd8c8bf621d108feb60de1f87437b803a

**Pull request merged**